### PR TITLE
[CAT-857] Move Storage to V2

### PR DESCRIPTION
**Pull request by nateshim-indico**

### CODE REVIEW

File: IndicoV2.IntegrationTests/GraphQLRequest/GraphQLRequestTests.cs [hunk 1]
```diff
@@ -34,8 +34,7 @@ public async Task RawQueryListDatasets_ShouldReturnResult()
             }";
             string operationName = "ListDatasets";
             dynamic variables = new { limit = 1 };
-            var result = await _graphQLRequestClient.Call(query, operationName, variables);
-
+            JObject result = await _graphQLRequestClient.Call(query, operationName, variables);
```
Comment (goatrocks) [inline IndicoV2.IntegrationTests/GraphQLRequest/GraphQLRequestTests.cs:37]: coding convention would usually have this be `var`

File: IndicoV2/Storage/MultipartFormUpload.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            string uploadUrl = _client.BaseUri + "/storage/files/store";
+            var formData = await MultipartFormDataContent(FileParameters);
+            var client = _client.HttpClient;
+            var responseMessage = await client.PostAsync(uploadUrl, formData, cancellationToken);
+            string body = await responseMessage.Content.ReadAsStringAsync();
+            var uploadResult = JArray.Parse(body);
+            return uploadResult;
+        }
+
+        private async Task<HttpContent> MultipartFormDataContent(List<FileParameter> parameters)
```
Comment (goatrocks) [inline IndicoV2/Storage/MultipartFormUpload.cs:31]: Hmm.. might be worth making a ticket and seeing if there's a better way to do this in the future, I feel like handling multipart form data boundaries maybe isn't something we should need to do by hand (not necessary to address for this task tho)

File: IndicoV2/TokenHandler.cs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+using System.Net.Http.Headers;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Indico.Exception;
+
+namespace IndicoV2
+{
+    internal class TokenHandler : DelegatingHandler
```
Comment (goatrocks) [inline IndicoV2/TokenHandler.cs:12]: This duplicates some of the code here: https://github.com/IndicoDataSolutions/indico-client-csharp/blob/7080fd610b7d01bae2377c0788dc3ea7176f8c0b/IndicoV2.StrawberryShake/HttpClient/AuthenticatingMessageHandler.cs#L12C32-L12C32

I wonder if it's feasible to refactor so we're using the HttpClient we inject into StrawberryShake? That way the auth code is re-used.

**Review by goatrocks — APPROVED** [review]
I left some comments -- particularly the auth/http client one (basically, investigate if we can use one httpclient for both strawberry shake + rest storage calls), if it's not huge to address, would ideally be addressed before submitting the PR. If it feels like more than a point or so of work, we can ticket it and address it in a future sprint, though.

File: IndicoV2.IntegrationTests/GraphQLRequest/GraphQLRequestTests.cs (hunk 1 above)
Comment (nateshim-indico) [reply IndicoV2.IntegrationTests/GraphQLRequest/GraphQLRequestTests.cs]: reverted

File: IndicoV2/TokenHandler.cs (hunk 3 above)
Comment (nateshim-indico) [reply IndicoV2/TokenHandler.cs:12]: putting an AuthenticationMessageHandler in worked!

**Pull request merged**